lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1637,6 +1637,56 @@ class DecomposeAtenIndexPutOp : public OpRewritePattern<AtenIndexPutOp> {
 };
 } // namespace
 
+
+namespace {
+// Decompose `aten.index_add` op into `valsem.aten.index_put_impl` op. with
+// accumulate set to True
+class DecomposeAtenIndexAddOp : public OpRewritePattern<AtenIndexAddOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenIndexAddOp op,
+                                PatternRewriter &rewriter) const override {
+
+    Value input = op.self();
+    int inputRank = getTensorRank(input);
+    BaseTensorType inputType = input.getType().cast<BaseTensorType>();
+    Value updates = op.source();
+
+    if (inputRank < 0) {
+      return rewriter.notifyMatchFailure(op, "Invalid Input: Unranked tensor");
+    }
+
+    if (getTensorRank(updates) != inputRank) {
+      return rewriter.notifyMatchFailure(
+          op, "Unimplemented: Ranks of input and updates do not match");
+    }
+
+    inputType = input.getType().cast<BaseTensorType>();
+    SmallVector<int64_t> empty;
+    Type expandUpdatesType = inputType.getWithSizesAndDtype(
+        llvm::makeArrayRef(empty), rewriter.getF64Type());
+
+    Value torchCstOne = rewriter.create<Torch::ConstantIntOp>(
+        op.getLoc(), rewriter.getI64IntegerAttr(1));
+
+    Value expandedUpdates = rewriter.create<AtenUnsqueezeOp>(
+        op.getLoc(), expandUpdatesType, updates, torchCstOne);
+
+    Value indicesAsTensorList = rewriter.create<PrimListConstructOp>(
+        op.getLoc(), Torch::ListType::get(updates.getType()),
+        Value(op.index()));
+
+    Value accumulateTrue = rewriter.create<Torch::ConstantBoolOp>(op.getLoc(), true);
+    Value cstFalse = rewriter.create<Torch::ConstantBoolOp>(op.getLoc(), false);
+
+    rewriter.replaceOpWithNewOp<ValsemVariantAtenIndexPutImplOp>(
```

Comment:
This decomposition seems to ignore the `dim` parameter of `torch.index_add`. AFAICT, there is no way to include that in `index_put`. So, how is this going to work when `dim != 0`? Am I missing something?

Reply:
But I don't see the indices being transformed to the form that you mention here. I believe that should be done using the dim value and I don't see the dim being used anywhere here.

Reply:
This PR is still a WIP, so there is indeed missing support currently for the `dim` argument.